draft-group-privacypass-k-check.md

```diff
@@ -62,23 +62,118 @@ The following terms are used throughout this document:
-here because it is generally bad practice to put content into GET requests, and POST
-responses can be cacheable.) Upon receipt, mirrors see if they have a cached version of
-the resource identified by the URL locally. If so, they return it to the client.
+The mirror protocol is a simple protocol similar to a reverse-proxy. Each mirror is identified
+by a Mirror URI Template {{!RFC6570}}. The scheme for the Mirror URI Template MUST be "https".
+The Mirror URI Template uses the Level 3 encoding defined {{Section 1.2 of RFC6570}} and contains
+one variables: "target", which is the percent-encoded URL of a HTTP resource to be mirrored.
+Example Mirror URI Templates are shown below.
+
+~~~
+https://mirror.example/mirror{?target}
+https://mirror.example/{target}
+~~~
+
+The Mirror URI Template MUST contain the "target" variable exactly once. The variable
+MUST be within the path or query components of the URI.
+
+Clients send requests to mirror resources after being configured with their corresponding Mirror
+URI Template. Clients MUST ignore configurations that do not conform to this template.
+
+Upon receipt of a mirror request, mirrors validate the incoming request. If the request is invalid
+or malformed, e.g., the "target" parameter is not a correctly encoded URL, the mirror aborts
+and returns a a 4xx (Client Error) to the client. Otherwise, if the request is valid, the mirror
+checks to see if it has a cached version of the resource identified by the target URL. If so, this
+response is returned to the client.
+
 Otherwise, mirrors send a GET request to the target resource URL. If this request fails,
-the mirror returns an error to the client. Otherwise, the response to a mirror request is
-the content that was contained in the target resource. With the exception of Cache-Control
-headers, the mirror response does not replicate any of the headers that may have been attached
-to the HTTP message carrying the target resource.
+the mirror returns a 4xx error to the client. Otherwise, the response to a mirror request is
+the content that was contained in the target resource. With the exception of Content-Type,
+Content-Length, and Cache-Control headers, the mirror response does not replicate any of the
+headers that may have been attached to the HTTP message carrying the target resource.
 
-[[OPEN ISSUE: Should there be some mandatory or RECOMMENDED minimum validity window of resources?]]
+Mirrors cache this content, indexed by the target URL, for future mirror requests. Mirrors
+purge this cache when the response is no longer valid according to the Cache-Control headers.
 
 We refer to the set of clients that interact with a mirror as mirror clients. We refer to
 the validity window of the mirror response as the period of time determined by the Cache-Control
 headers as the response.
 
+[[OPEN ISSUE: Should there be some mandatory or RECOMMENDED minimum validity window of resources?]]
```

Comment:
I wonder if the mirror should just be enforcing this. It doesn't seem possible for the mirror to enforce max validity windows without making these decisions locally (it can't tell max-age=10 seconds means that the resource caching is too low or it has just hit the resource towards the end of its lifetime.

Reply:
Yeah, maybe. Let's track this in #1.